### ISSUE TITLE
failed to deploy prometheus rule

### ISSUE THREAD

**Issue by ilovekate19** [rook/rook#10022]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**
2022-04-08 00:07:08.001314 E | op-mgr: failed to deploy prometheus rule. failed to set owner reference to prometheus rule "prometheus-ceph-v16-rules": cross-namespace owner references are disallowed, owner's namespace rook-ceph, obj's namespace monitoring

**Expected behavior:**
prometheus rule deployed on monitoring namespace not rook-ceph.

**How to reproduce it (minimal and precise):**

Prometheus is deployed on monitoring namespace and rook-ceph on rook-ceph namespace.
I managed to check servicemonitor shown up on prometheus UI but rules are not there.
Here is rook-ceph cluster.yaml which deployed by helm.
```
# rook-ceph cluster.yaml

toolbox:
  enabled: true


cephClusterSpec:
  placement:
    all:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
            - matchExpressions:
              - key: cephnode
                operator: In
                values:
                - enabled
    mon:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
            - matchExpressions:
              - key: cephmon
                operator: In
                values:
                - enabled

cephBlockPoolsVolumeSnapshotClass:
  enabled: true
  name: ceph-block
  parameters:
    clusterID: rook-ceph
    csi.storage.k8s.io/snapshotter-secret-name: rook-csi-rbd-provisioner
    csi.storage.k8s.io/snapshotter-secret-namespace: rook-ceph  

cephFileSystemVolumeSnapshotClass:
  enabled: true
  name: ceph-filesystem
  parameters:
    clusterID: rook-ceph
    csi.storage.k8s.io/snapshotter-secret-name: rook-csi-cephfs-provisioner
    csi.storage.k8s.io/snapshotter-secret-namespace: rook-ceph    

monitoring:
  enabled: true
  createPrometheusRules: true
  rulesNamespaceOverride: monitoring

```
operator.yaml
```
nodeSelector:
  cephnode: enabled
csi:
  provisionerNodeAffinity: cephnode=enabled
monitoring:
  enabled: true

```

And operator got this error.
```
2022-04-08 00:07:08.001314 E | op-mgr: failed to deploy prometheus rule. failed to set owner reference to prometheus rule "prometheus-ceph-v16-rules": cross-namespace owner references are disallowed, owner's namespace rook-ceph, obj's namespace monitoring
```
prometheus-ceph-v16-rules deployed on rook-ceph not monitoring namespace.
$ k get prometheusrules 
NAMESPACE    NAME                                                              AGE
rook-ceph    prometheus-ceph-v16-rules                                         15h

**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

 To get logs, use `kubectl -n <namespace> logs <pod name>`
When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
Read [GitHub documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Environment**:
* OS (e.g. from /etc/os-release):ubuntu 20.04
* Kernel (e.g. `uname -a`):Linux hqpcdworker1 5.4.0-100-generic #113-Ubuntu SMP Thu Feb 3 18:43:29 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
* Cloud provider or hardware configuration: on prem
* Rook version (use `rook version` inside of a Rook Pod):
```
$ helm ls
NAME                    NAMESPACE       REVISION        UPDATED                                 STATUS          CHART                           APP VERSION
rook-ceph               rook-ceph       4               2022-04-07 17:29:57.32768495 +0900 KST  deployed        rook-ceph-v1.8.7                v1.8.7
rook-ceph-cluster       rook-ceph       13              2022-04-08 09:04:18.851087453 +0900 KST deployed        rook-ceph-cluster-v1.8.7        v1.8.7
```
* Storage backend version (e.g. for ceph do `ceph -v`):
* Kubernetes version (use `kubectl version`):
```
$ k version
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.4", GitCommit:"e6c093d87ea4cbb530a7b2ae91e54c0842d8308a", GitTreeState:"clean", BuildDate:"2022-02-16T12:38:05Z", GoVersion:"go1.17.7", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.6+rke2r1", GitCommit:"f59f5c2fda36e4036b49ec027e556a15456108f0", GitTreeState:"clean", BuildDate:"2022-01-19T21:17:43Z", GoVersion:"go1.16.12b7", Compiler:"gc", Platform:"linux/amd64"}
```
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): RKE2
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/latest/ceph-toolbox.html)):
```
[rook@rook-ceph-tools-588f8d8966-7gqpb /]$ ceph status
  cluster:
    id:     93bfc186-9f9a-4be4-b9d9-5f7a91d06176
    health: HEALTH_WARN
            clock skew detected on mon.b, mon.c

  services:
    mon: 3 daemons, quorum a,b,c (age 2w)
    mgr: a(active, since 17h)
    mds: 1/1 daemons up, 1 hot standby
    osd: 3 osds: 3 up (since 2w), 3 in (since 2w)
    rgw: 1 daemon active (1 hosts, 1 zones)

  data:
    volumes: 1/1 healthy
    pools:   11 pools, 177 pgs
    objects: 9.01k objects, 32 GiB
    usage:   98 GiB used, 446 GiB / 544 GiB avail
    pgs:     177 active+clean

  io:
    client:   3.2 KiB/s rd, 24 KiB/s wr, 4 op/s rd, 4 op/s wr

[rook@rook-ceph-tools-588f8d8966-7gqpb /]$ ceph health
HEALTH_WARN clock skew detected on mon.b, mon.c
```

**Comment by travisn** [comment]
While #10028 will fix this issue in the next v1.8.x release, note that this issue will not apply in the v1.9 that is due out in the next few days. The new approach will create the prometheus rules purely from the helm chart and will never set the ownerref even if in the same namespace as the operator. The helm chart will fully own the rules rather than the cephcluster.

**Comment by ilovekate19** [comment]
Thanks @travisn for your quick clarification. 
FYI, I used a workaround now by putting additional rule on prometheus helm chart.
Let me close this after new release comes.
